akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -941,35 +940,69 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
   /**
    * Initiates a new round of gossip.
    */
-  def gossip(): Unit = {
-
+  def gossip(): Unit =
     if (!isSingletonCluster) {
-      val localGossip = latestGossip
+      val gossipTargets =
+        // TODO break this two methods out and unit test?
+        if (latestGossip.isMultiDc) multiDcGossipTargets()
+        else localDcGossipTargets()
+
+      selectRandomNode(gossipTargets) match {
```

Comment:
Thought about it some more and now I agree that choosing a DC randomly first is better because of not preferring DCs with more nodes than the limit over DCs with fewer nodes. I'll update the logic.